scripts/get_capella_components.js

```diff
@@ -0,0 +1,288 @@
+      analyzeImports(source);
+    }
+    return;
+  }
+  const files = fs.readdirSync(source);
+
+  // Loop through the files and directories in the source directory
+  files.forEach((file) => {
+    const sourceFilePath = path.join(source, file);
+    const destinationFilePath = path.join(destination, file);
+
+    // Check if the current item is a file or directory
+    const isDirectory = fs.statSync(sourceFilePath).isDirectory();
+
+    if (isDirectory) {
+      // If it's a directory, recursively copy its contents
+      copyDirectory(sourceFilePath, destinationFilePath);
+    } else {
+      if (!excludedExtensions.some((ext) => sourceFilePath.endsWith(ext))) {
+        // If not excluded, copy the file to the destination directory
+        fs.copyFileSync(sourceFilePath, destinationFilePath);
+        analyzeImports(sourceFilePath, destinationFilePath);
+      }
+    }
+  });
+} catch(err)
+{
+  console.log(err);
+}
+};
+
+const componentImports = [];
+const componentImported = [];
+const externalDependencies = [];
+
+
+const doesFileExist = (filePath) => {
+  try {
+    const stats = fs.statSync(filePath);
+    return stats.isFile(); // Returns true if it's a file, false if it's a directory or does not exist.
+  } catch (err) {
+    return false; // File does not exist or there was an error accessing it.
+  }
+};
+
+const isLastTwoComponentsSame = (pathString) => {
+  const components = pathString.split('/');
+  if (components.length < 2) {return false;}
+  const lastComponent = components.pop();
+  const secondLastComponent = components.pop();
+  return lastComponent === secondLastComponent;
+};
+
+// Function to read file and analyze import statements
+const analyzeImports = (filePath, destinationPath) => {
+  const fileContent = fs.readFileSync(filePath, 'utf8');
+  const ast = ts.createSourceFile(filePath, fileContent, ts.ScriptTarget.Latest, true);
+
+  const visitNode = (node) => {
+    if (ts.isImportDeclaration(node)) {
+      let importPath = node.moduleSpecifier.text;
+
+      if(isLastTwoComponentsSame(importPath))
+      {
+        importPath = path.dirname(importPath);
+      }
+      const sourceComponentPath = path.join(pathToCapella, importPath + ".ts");
+      if(doesFileExist(sourceComponentPath))
+      {
+        const directoryName = path.dirname(importPath);
+        if(directoryName === "utils" || directoryName === "constants" || directoryName === "types" || directoryName === "error" || directoryName === "sync" ) {
+          importPath = importPath + ".ts";
+        }
+        else {
+        importPath = path.dirname(importPath);
+        }
+      }
+      if (importPath.startsWith('components') || importPath.startsWith('hooks') || importPath.startsWith('types') || importPath.startsWith('utils') || importPath.startsWith('sync') || importPath.startsWith('.') || importPath.startsWith('constants') || importPath.startsWith('error')) {
+        if(importPath.startsWith('.'))
```

Comment:
Can we refactor it to if(!importPath.startsWith('.') &&((importPath.endsWith('utils') && .....)?